src/structlog/_base.py

```diff
@@ -113,39 +97,42 @@ def try_unbind(self, *keys):
 
         return bl
 
-    def new(self, **new_values):
+    def new(self, **new_values: Any) -> "BoundLoggerBase":
         """
-        Clear context and binds *initial_values* using :func:`bind`.
+        Clear context and binds *initial_values* using `bind`.
 
         Only necessary with dict implementations that keep global state like
         those wrapped by `structlog.threadlocal.wrap_dict` when threads
         are re-used.
-
-        :rtype: ``self.__class__``
         """
         self._context.clear()
+
         return self.bind(**new_values)
 
     # Helper methods for sub-classing concrete BoundLoggers.
 
-    def _process_event(self, method_name, event, event_kw):
+    def _process_event(
+        self, method_name: str, event: Optional[str], event_kw: Dict[str, Any]
```

Comment:
for accepting parameters it's usually better to use `Mapping` instead of `Dict` as it's more flexible to callers (though, if this is an internal method (which it looks like) then concrete types are fine).  (something something covariance, I forget the precise bit of type theory that covers this)

Reply:
I tend to use Mapping, however process_event needs a `copy()` which is not part of it. I may define a more hand-tailored Protocol for it but unfortunately `class CopyableMapping(Mapping, Protocol)` doesn't work because Mapping is not a Protocol.

Reply:
oof right 🤦 -- could use `copy.copy(...)` but sigh

Reply:
I actually played around with it but ultimately decided it's not worth it. I'm gonna say it again, that structlog is a terrible project for static types. ;)